### PR TITLE
Bump Aura.Intl version to 3.0.

### PR DESCRIPTION
**Pull request by ADmad**
Refs https://github.com/auraphp/Aura.Intl/releases/tag/3.0.0-beta1

Before we decide to go ahead with the newer version it would be better to know what breaking API changes, if any, they plan for Aura.Intl 3.0.

### CODE REVIEW

**Comment by dereuromark** [comment]
Looks good, could we already use `>=1.1` etc to support both for the time being?

**Comment by markstory** [comment]
@dereuromark Couldn't that be `1.1.* | 3.0.*`?

**Comment by dereuromark** [comment]
Is 2.0 not a good idea? but sure, that's also fine probably.

**Comment by ADmad** [comment]
There isn't and won't be any 2.x of Aura.Intl  https://github.com/auraphp/Aura.Intl/issues/33

File: composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "ext-intl": "*",
         "ext-mbstring": "*",
         "cakephp/chronos": "~1.0",
-        "aura/intl": "1.1.*",
+        "aura/intl": "3.0.*",
```
Comment (harikt) [inline composer.json]: why not use 3.0.*@beta so only aura/intl will install beta.

**Comment by harikt** [comment]
When is 3.4 getting released ? So we can make a stable release of Aura.Intl before that. There is no public API breaks as you are using it as standalone. @lorenzo has tested this months back . Only thing is I made php version 5.6+ constraint for composer.json. But if you guys need to change the composer.json to 5.5 + to support 3.x of cakephp, we can also do the same. See discussion https://groups.google.com/d/msg/auraphp/GdZlkaCzeZo/rsPAiYT0EQAJ .

File: composer.json (hunk 1 above)
Comment (ADmad) [inline composer.json]: Cause my composer versioning game is weak :stuck_out_tongue:

**Comment by markstory** [comment]
> When is 3.4 getting released ? 

@harikt I'm going to to the beta release later this week, aiming to have a stable release mid January.

**Comment by harikt** [comment]
@markstory cool. I believe we can do a stable release before that.

Let me know if you guys run into any issues.

Thank you.

File: composer.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
         "ext-intl": "*",
         "ext-mbstring": "*",
         "cakephp/chronos": "~1.0",
-        "aura/intl": "1.1.*",
+        "aura/intl": "3.0.*@beta",
```
Comment (harikt) [inline composer.json]: Released a stable today ;-) . https://github.com/auraphp/Aura.Intl/releases/tag/3.0.0

**Comment by lorenzo** [comment]
@ADmad Tests seem to be failing with translator stuff

**Comment by ADmad** [comment]
@lorenzo They weren't when I initially made the PR. I'll check.

**Comment by ADmad** [comment]
The error is caused due to change in signature of [Translator::__construct()](https://github.com/auraphp/Aura.Intl/blob/3.x/src/Translator.php#L72).  In 1.x the second param is [array $message](https://github.com/auraphp/Aura.Intl/blob/1.x/src/Aura/Intl/Translator.php#L73). 

We added a custom `Translator` class recently in #10004 to fix issues with context and plurals.

Just skimming through our `Translator` class it seems it can be easily updated.

**Comment by harikt** [comment]
@ADmad sorry that I missed it to document in the changelog.

File: src/I18n/Translator.php [hunk 3]
```diff
@@ -52,28 +53,28 @@ class Translator implements TranslatorInterface
...
      * @param string $locale The locale being used.
-     * @param array $messages The message keys and translations.
+     * @param \Aura\Intl\Package $package The Package containing keys and translations.
      * @param \Aura\Intl\FormatterInterface $formatter A message formatter.
      * @param \Aura\Intl\TranslatorInterface|null $fallback A fallback translator.
      */
     public function __construct(
         $locale,
-        array $messages,
+        Package $package,
```
Comment (markstory) [inline src/I18n/Translator.php]: I'm glad I marked this class as 'internal'.

**Comment by markstory** [comment]
Looks good to me :+1:

**Comment by ADmad** [comment]
@lorenzo Good to go?

**Comment by dereuromark** [comment]
👍

**Pull request merged**